src/Stratis.Bitcoin.Features.Dns/DnsFeature.cs

```diff
@@ -117,14 +135,17 @@ public override Task InitializeAsync()
             this.dnsTask = Task.Factory.StartNew(this.RunDnsService, TaskCreationOptions.LongRunning);
 
             this.StartWhitelistRefreshLoop();
-            
+
+            IServiceProvider serviceProvider = this.services.ServiceProvider;
+            NetworkPeerConnectionParameters connectionParameters = this.connectionManager.Parameters;
+            connectionParameters.TemplateBehaviors.Add(serviceProvider.GetService<UnreliablePeerBehavior>());
```

Comment:
`UnreliablePeerBehavior` is injected so can be pass in constructor.

Reply:
Fine, I'm then removing all references to to `services` member for the moment, so we inject directly the behavior into the constructor.